src/Views/HotspotPage.vala

```diff
@@ -131,13 +133,12 @@
         }
 
         private void validate_entries () {
-            bool key_text_over_8 = key_entry.text.length >= 8;
-            status_switch.sensitive = key_text_over_8 || !sensitive;
-
-            if (!key_text_over_8 && key_entry.text != "") {
-                key_entry.set_icon_from_icon_name (Gtk.EntryIconPosition.SECONDARY, "process-error-symbolic");
-            } else {
-                key_entry.set_icon_from_icon_name (Gtk.EntryIconPosition.SECONDARY, null);
+            status_switch.sensitive = key_entry.is_valid || !sensitive;
+            if (!key_entry.is_valid) {
+                    key_entry.secondary_icon_tooltip_text = _("Password needs to be at least 8 characters long");
+                }
+            else {
+                key_entry.secondary_icon_tooltip_text = _("Password is at least 8 characters long");
```

Comment:
Since Granite.ValidatedEntry uses a success icon I found it better to have a tooltip pointing that the success only applies to the 8 character minimum requirement and doesn't mean anything else like password strength.